templates/workspaces/base/terraform/storage.tf

```diff
@@ -8,6 +8,12 @@ resource "azurerm_storage_account" "stg" {
   lifecycle { ignore_changes = [tags] }
 }
 
+resource "azurerm_storage_share" "shared_storage" {
+  name                 = "vm-shared-storage"
+  storage_account_name = azurerm_storage_account.stg.name
+  quota                = 50
+}
+
```

Comment:
This is probably not the perfect place to comment about this, but, this storage account will now be "compromised", as in, can't be used for anything else since its key is visible from each vm.
1. Are we using it for anything else?
2. Perhaps we should make it more visible to a developer to be careful when using it? Names, comments, maybe delete the blob private endpoint since that probably not a valid use for this account?

Reply:
The storage account referenced is the one created as part of a base workspace deployment.  Prior to the changes in this PR this storage account was not in use.  Part of this PR creates a File Share in the SA which is then made available for access to the VMs within Guacamole.

I had originally tried alternative methods of mounting the drives on both Linux and Windows although all failed to persist the drive or execute correctly.

Reply:
As the workspace is seen as the security boundary I think this is acceptable, and not sure is any other way to avoid this from a technical perspective.

The requirement I see from customers is generally when data is within the workspace any researcher should be free to access that data.

We should have some documentation calling this out. Part of the base workspace template maybe?